collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -762,7 +762,7 @@ protected void onActivityResult(int requestCode, int resultCode, final Intent in
 
         // If we're coming back from the hierarchy view, the user has either tapped the back
         // button or another question to jump to so we need to rebuild the view.
-        if (requestCode == RequestCodes.HIERARCHY_ACTIVITY) {
+        if (requestCode == RequestCodes.HIERARCHY_ACTIVITY || requestCode == RequestCodes.CHANGE_SETTINGS) {
```

Comment:
Can we do the same for font size? Probably we don't even need adjustFontSize() now if we call onScreenRefresh(); either way. Do you want to confirm? If you are in a hurry with this pr we can file a separate issue to make sure changes in settings are handled in the same way.

Reply:
Sounds like a good idea! Just to confirm: you're suggesting we get rid of the code that's dealing with the font size adjustment and see if this new solution works instead?

Reply:
Yeah I guess it should because basically it does the same just calling onScreenRefresh();